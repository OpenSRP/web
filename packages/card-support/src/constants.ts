// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
