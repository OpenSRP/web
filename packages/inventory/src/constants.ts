// opensrp api strings
export const OPENSRP_API_BASE_URL = 'https://mg-eusm-staging.smartregister.org/opensrp/rest/';

// router routes
export const INVENTORY_SERVICE_POINT_LIST_VIEW = '/inventory/list';
export const INVENTORY_SERVICE_POINT_PROFILE_VIEW = '/inventory/list';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'inventory';

// magic strings
export const SEARCH_QUERY_PARAM = 'querySearch';
export const TABLE_PAGE_SIZE = 5;
export const TABLE_PAGE_SIZE_OPTIONS = ['5', '10', '20', '50', '100'];

export const tablePaginationOptions = {
  showQuickJumper: true,
  showSizeChanger: true,
  defaultPageSize: TABLE_PAGE_SIZE,
  pageSizeOptions: TABLE_PAGE_SIZE_OPTIONS,
};

//endpoints
export const GET_INVENTORY_BY_SERVICE_POINT = 'stockresource/servicePointId/';

//payload
export const inventory1 = {
  stockId: '5dfcaecd-430a-4284-972f-9acf87775a9b1',
  productName: 'Scale1',
  unicefSection: 'Health1',
  quantity: 4,
  deliveryDate: '2020-02-01',
  accountabilityEndDate: '2020-10-10',
  donor: 'ADB',
  servicePointId: '903971',
  poNumber: 1111,
  serialNumber: '1234serial2',
  providerId: 'providerId',
};

export const inventory2 = {
  stockId: '5dfcaecd-430a-4284-972f-9acf87775a9b2',
  productName: 'Scale2',
  unicefSection: 'Health2',
  quantity: 4,
  deliveryDate: '2020-02-01',
  accountabilityEndDate: '2020-10-10',
  donor: 'ADB',
  servicePointId: '903972',
  poNumber: 1112,
  serialNumber: '1234serial2',
  providerId: 'providerId',
};
