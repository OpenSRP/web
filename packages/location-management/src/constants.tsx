// URLs
export const URL_ALL_LOCATION_TAGS = '/location-tag';
export const URL_DELETE_LOCATION_TAGS = '/location-tag/delete';
export const KEYCLOAK_API_BASE_URL ='https://opensrp-stage.smartregister.org/opensrp/rest';
