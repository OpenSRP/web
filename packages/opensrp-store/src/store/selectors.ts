import { reducerName as sessionReducerName } from '@onaio/session-reducer';
import { Store } from 'redux';
import { Dictionary } from '@onaio/utils';
import { createSelector } from 'reselect';

/** get API Token from the Redux store
 *
 * @param {any} state - the redux store
 */
interface StoreState {
  session: {
    [key: string]: Dictionary;
  };
}

interface APIFilters {
  accessToken?: boolean;
  apiToken?: boolean;
  providerState?: boolean;
}

/** Fetch access token from store if access_token filter val is true
 *
 * @param _ - redux store
 * @param props - api filters object
 * @returns {boolean}
 */

export const fetchAccessToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.accessToken || null;

/** Fetch api token from store if api_token filter val is true
 *
 * @param _ redux store
 * @param props api filters object
 * @returns {boolean}
 */

export const fetchApiToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.apiToken || null;

/** Fetch oauth provider state from store if providerState filter val is true
 *
 * @param _ redux store
 * @param props api filters object
 * @returns {boolean}
 */

export const fetchOauthProviderState = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.providerState || null;

/** Gets extra data object from store
 *
 * @param state - redux store
 */

export const getExtraData = (state: Partial<Store>): Dictionary =>
  (state as StoreState)[sessionReducerName].extraData;

/**
 * @param {any} state - the redux store
 * @returns {Object} - the states
 */

export const makeAPIStateSelector = () =>
  createSelector(
    [fetchAccessToken, fetchApiToken, fetchOauthProviderState, getExtraData],
    (accessToken, apiToken, oauthState, extraData) => {
      if (accessToken) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.access_token) || null;
      } else if (apiToken) {
        return extraData.api_token || null;
      } else if (oauthState) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.state) || null;
      }
    }
  );
