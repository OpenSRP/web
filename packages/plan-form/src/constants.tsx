export const A1 = 'A1';
export const A2 = 'A2';
export const B1 = 'B1';
export const B2 = 'B2';
export const LOW_PRIORITY = 'low-priority';
export const MEDIUM_PRIORITY = 'medium-priority';
export const HIGH_PRIORITIY = 'high-priority';
export const TRUE = 'True';
export const FALSE = 'False';
export const DISABLED = 'Disabled';
export const INTERNAL = 'internal';
export const IGNORE = 'ignore';
export const INTERVENTION_TYPE_CODE = 'interventionType';
export const FI_STATUS_CODE = 'fiStatus';
export const FI_REASON_CODE = 'fiReason';
export const OPENSRP_EVENT_ID_CODE = 'opensrpEventId';
export const CASE_NUMBER_CODE = 'caseNum';
export const TASK_GENERATION_STATUS_CODE = 'taskGenerationStatus';
export const TEAM_ASSIGNMENT_STATUS_CODE = 'teamAssignmentStatus';
export const CASE_TRIGGERED = 'Case Triggered';
export const ROUTINE = 'Routine';
export const INVESTIGATION = 'Investigation';
export const DAYS = 'days';
export const CONDITION = 'condition';
export const TRIGGER = 'trigger';

// plan activity codes
export const CASE_CONFIRMATION_ACTIVITY_CODE = 'caseConfirmation';
export const FAMILY_REGISTRATION_ACTIVITY_CODE = 'familyRegistration';
// opensrp api strings
export const PLAN_LIST_URL = '/plans/list';
export const OPENSRP_API_BASE_URL = '';
export const OPENSRP_PLAN_ENDPOINT = 'plans';