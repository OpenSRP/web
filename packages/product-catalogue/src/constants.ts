// opensrp api strings
export const OPENSRP_PRODUCT_CATALOGUE = 'rest/product-catalogue';

// router routes
export const CATALOGUE_LIST_VIEW_URL = '/product-catalogue';
export const CATALOGUE_EDIT_VIEW_URL = '/product-catalogue/edit';
export const CATALOGUE_CREATE_VIEW_URL = 'product-catalogue/new';
export const HOME_URL = '/';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'product-catalogue';

/** route params for product-catalogue pages */
export interface RouteParams {
  productId?: string;
}
