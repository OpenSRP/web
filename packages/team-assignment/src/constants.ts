export const OPENSRP_API_BASE_URL = 'https://opensrp-staging.smartregister.org/opensrp/rest/';

// router routes
export const TEAM_ASSIGNMENT_LIST_VIEW_URL = '/team-assignment';
export const TEAM_ASSIGNMENT_EDIT_VIEW_URL = '/team-assignment/edit';
export const TEAM_ASSIGNMENT_CREATE_VIEW_URL = 'team-assignment/new';
export const HOME_URL = '/';
export const LOCATION_HIERARCHY_ENDPOINT = 'location/hierarchy';
export const PLANS_ENDPOINT = 'plans';
export const ASSIGNMENTS_ENDPOINT = 'organization/assignLocationsAndPlans';
export const SUCCESSFULLY_ASSIGNED_TEAMS = 'Successfully Assigned Teams';
export const ENTER_TEAM_NAME = 'Enter a Team name';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'team-assignment';

// Table view strings

export const TEAM_ASSIGNMENT_PAGE_TITLE = 'Team Assignment';
export const SAVE = 'Save';
export const CANCEL = 'Cancel';
