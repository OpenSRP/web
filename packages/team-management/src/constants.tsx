export const API_BASE_URL = process.env.REACT_APP_OPENSRP_API_BASE_URL;

//url
export const URL_ADD_TEAM = 'teams/add';
export const URL_EDIT_TEAM = 'teams/edit/';

//
// Teams detail
//
export const TEAMS_GET = 'organization';

export const PRACTITIONER_GET = 'practitioner';
export const PRACTITIONER_POST = 'practitionerRole/add';
export const TEAM_PRACTITIONERS = 'organization/practitioner/';
export const PRACTITIONER_ROLE_ADD = 'practitionerRole/add';

export const TEAM_UUID = 'team-uuid';

export const ERROR_OCCURRED = 'An error occurred';
