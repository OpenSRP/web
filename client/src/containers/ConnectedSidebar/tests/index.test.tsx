import { authenticateUser } from '@onaio/session-reducer';
import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import ConnectedSidebar from '..';
import { store } from '@opensrp/store';
import { MISSIONS } from '../../../constants';

describe('components/ConnectedSidebar', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the ConnectedSidebar component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('aside').props()).toMatchSnapshot();
  });

  it('renders user managment menu for users with appropriate role', () => {
    store.dispatch(
      authenticateUser(
        true,
        {
          email: 'bob@example.com',
          name: 'Bobbie',
          username: 'RobertBaratheon',
        },
        {
          roles: ['ROLE_EDIT_KEYCLOAK_USERS'],
          username: 'superset-user',
          user_id: 'cab07278-c77b-4bc7-b154-bcbf01b7d35b',
        }
      )
    );
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
  });

  it('Test order of menu', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_PRODUCT_CATALOGUE = 'true';
    envModule.ENABLE_PLANS = 'true';
    envModule.ENABLE_LOCATIONS = 'true';
    envModule.ENABLE_FORM_CONFIGURATION = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find(`Menu`).last().prop('children')).toMatchSnapshot();
  });

  it('displays menu links for enabled Plans module', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_PLANS = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find(`SubMenu[title="${MISSIONS}"]`)).toHaveLength(1);
  });

  it('displays menu links for enabled Product Cataglogue module', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_PRODUCT_CATALOGUE = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );

    const x = (wrapper
      .find('SubMenu[title="Admin"]')
      .first()
      .last()
      .prop('children') as ReactWrapper[])[2].key;

    expect(x).toMatch('product-catalogue');
  });

  it('displays menu links for enabled Location module', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_LOCATIONS = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );

    const x = (wrapper
      .find('SubMenu[title="Admin"]')
      .first()
      .last()
      .prop('children') as ReactWrapper[])[3].key;

    expect(x).toMatch('admin-locations');
  });

  it('displays menu links for enabled Form Configuration module', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_FORM_CONFIGURATION = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );

    const x = (wrapper
      .find('SubMenu[title="Admin"]')
      .first()
      .last()
      .prop('children') as ReactWrapper[])[4].key;

    expect(x).toMatch('admin-form-config');
  });
});
