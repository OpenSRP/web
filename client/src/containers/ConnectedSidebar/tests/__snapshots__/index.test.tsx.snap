// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ConnectedSidebar Test order of menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configurations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand admin-form-config menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configurations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand form-config menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configurations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand location menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configurations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand location menu 2`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configurations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand users menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configurations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand users menu 2`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configurations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;
